HIP/hip-655.md

```diff
@@ -0,0 +1,106 @@
+---
+hip: 655
+title: Token airdrops without explicit associations 
+author: Nick Poorman <@nickpoorman>
+working-group: Richard Bair <@rbair23>, Jasper Potts <@jasperpotts>, Atul Mahamuni <@atul-hedera>, Michael Tinker <@tinker-michaelj>, Leemon Baird <@lbaird>
+type: Standards Track 
+category: Service
+needs-council-approval: Yes 
+status: Review
+created: 2022-10-20 
+discussions-to: https://github.com/hashgraph/hedera-improvement-proposal/discussions/654
+updated: 2023-01-20, 2023-01-24
+---
+
+## Abstract
+
+[HIP-23](https://hips.hedera.com/hip/hip-23) allowed an account owner to open up a specified number of automatic token-association slots into which others can transfer tokens. However, this capability doesn't allow for a pure "airdrop" of tokens. This HIP provides an additional mechanism for senders to airdrop tokens to other accounts, while allowing individual account holders to maintain control over what shows up in their account balances.
+
+## Motivation
+
+The current implementation of the Hedera Token Service does not allow airdrops of arbitrary tokens to any account. [HIP-23](https://hips.hedera.com/hip/hip-23) lists several disadvantages of arbitrary airdrops - such as exposing an innocent account holder to reputational or legal or spam risk of being associated with unwanted/unsolicited tokens, as well as DDoS risks for the network. And it proposes opt-in token associations. Further, [HIP-367](https://hips.hedera.com/hip/hip-367) removes the limit on token associations. Even with these enhancements, users still cannot airdrop tokens using the Hedera network today.
+
+## Rationale
+
+Token airdrops are very common in the blockchain industry and many legitimate use cases require airdropping of tokens. This HIP attempts to provide a mechanism to enable these airdrops while still protecting the account owners, and imposing dis-incentivizing costs to the senders of the airdropped tokens.
+
+## User Stories
+
+### Terminology
+
+The following terminology is used in the rest of this Specification
+
+- Airdrop-Sender: the owner of a token that they wish to send to somebody without the explicit association/permission from the intended recipient
+- Airdrop-Recipient: the intended recipient of the above airdrop
+- Node-Operator: the operator of a node in the Hedera network
+    
+### User stories
+    
+- As an Airdrop-Sender, I want to send arbitrary tokens (Fungible and/or Non-fungible) to Airdrop-Recipients so that I can create interesting business use cases
+    1. I understand that the recipients of the airdrop may not wish to accept these tokens and I am willing to pay for the network resources used while they make their decision.
+    2. If the intended Airdrop-Recipients do not accept my transfer, I want to be able to restore those tokens to my account so that I can do whatever I wish with those tokens.
```

Comment:
3. I want to be able to query the status of my airdrops, whether they are : Success , Pending (and the upcoming renewal fees/date), Expired or Rejected.
4. I want to set the number of renewal cycles that I'm willing to pay for, and have the airdrop expire if that limit is reached.

Reply:
Scheduled transactions has that covered.

Reply:
> 3\. I want to be able to query the status of my airdrops, whether they are : Success , Pending (and the upcoming renewal fees/date), Expired or Rejected.

Seems reasonable. I think this would be covered under the query to mirror node. Although it may take a few different requests to Mirror Node to gather all that data; probably would be fewer requests with something like a GraphQL API. With the `unsolicitedInbox` & `unsolicitedOutbox` queries, you should be able to deduce the Pending, Success, and Rejected status. Expired status would probably come from querying the existing Mirror Node APIs. In any case, it would be extremely simple to aggregate and cache a rolled up view of this data.

> I want to set the number of renewal cycles that I'm willing to pay for, and have the airdrop expire if that limit is reached.

I'd have to think about this one more.